src/components/Button/Button.tsx

```diff
@@ -92,6 +92,8 @@ export interface ButtonProps extends VibeComponentProps {
   insetFocus?: boolean;
   /** Specifies the tab order of an element */
   tabIndex?: number;
+  /** Controls resize observer for smooth transition between states. Do not use with loading state */
```

Comment:
maybe describe a bit more why it is needed

Reply:
1. I’d name the prop in a meaningful way which doesn’t explain the technology (resize observer) but rather what is the expected behavior (dynamic width?)
 2. It’s worth documenting such things in stories even though they are mostly technical, developers want to see how the code affects the design
 3. If we wish to make this the default behavior, we should start introducing a deprecation mechanism the console-logs the updated behavior. For example, if in the next major version we would like to eliminate it completely - a depredation log should say that in the next major version this prop will be turned on by default and we suggest migrating already to the new behavior. If the case is that it can be turned on/off - then the message should say what would be it’s default value (true?) and ask to pass explicitly the wanted behavior (true/false).

Reply:
> I’d name the prop in a meaningful way which doesn’t explain the technology (resize observer) but rather what is the expected behavior (dynamic width?)

I agree the prop name should be meaningful, but it aims to fix an existing bug that is related to the resize observer. For example either way the button will have a dynamic width, but in combination with the loading state there is the bug described (where the button doesn't shrink), maybe `disableDynamicResizing`? It might also mislead I think

> It’s worth documenting such things in stories even though they are mostly technical, developers want to see how the code affects the design

I agree, I added a JSDoc comment so it will appear in the props table, but I'm wondering how would we be able to describe such case in a story, do you have any idea? Because having the new prop as true in addition to `loading` will result in a bad example of the UI

> If we wish to make this the default behavior, we should start introducing a deprecation mechanism the console-logs the updated behavior. For example, if in the next major version we would like to eliminate it completely - a depredation log should say that in the next major version this prop will be turned on by default and we suggest migrating already to the new behavior. If the case is that it can be turned on/off - then the message should say what would be it’s default value (true?) and ask to pass explicitly the wanted behavior (true/false).

Sounds cool, actually we wanted to remove the resize observer completely in the next major version and solve this case somehow with CSS only, so this prop will be deprecated hopefully

Reply:
I’ve looked into your PR description again and suggest to name it “disableLoaderSmoothing” in case we wish to keep the bad behavior as default, or “loaderSmoothing” if we want to break the current behavior.

I kind of tending towards removing this logic completely if it makes more problems than benefits